# **Block 1: Import libraries**

In [ ]:
import numpy as np
import pandas as pd

# **Block 2: Load the data**

In [ ]:
data = pd.read_csv('problem_set_1_data.csv')

# rename the columns
data = data.rename(columns = {"quizzes": "quiz_total", "attendedClass": "attended", "finalgrade": "final_grade", "final_EXAM": "final_exam", "student": "name"})

# print the head
print('head: ')
print(data.head())
print('')  #skip a line in the output so it looks neater

#print the tail
print('tail:')
print(data.tail())
print('')

# print a summary of the dataframe
print('summary: ')
print(data.describe())

head: 
   quiz_total  attended  final_grade  final_exam               name
0       78.50         1        99.38          46       Linnie Lietz
1       64.25         0        95.78          45      Mitch Mustain
2       76.75         0        97.42          41     Salina Chavera
3       69.00         1        92.50          38  Kimberely Conwell
4       64.00         0        92.50          39          Zack Burk

tail:
     quiz_total  attended  final_grade  final_exam                 name
166        73.5         0        96.25          45           Dana Malta
167        75.5         0        98.13          44         Sixta Heyden
168        41.0         0        69.38          32       Brianne Broome
169        77.0         0       101.25          47  Elouise Weatherholt
170        72.0         0        92.81          38        Addie Maharaj

summary: 
       quiz_total    attended  final_grade  final_exam
count  171.000000  171.000000   171.000000  171.000000
mean    69.131579    0.19

### The averages for each column show the average scores that were received in the class on the summed quiz grades, the average attendance (based on either 1 for attendance and 0 for no attendance), the average final grade, and the average score on the final exam.

# **Block 3: Subset the data by whether someone attended class**

In [ ]:
attendedclass = data['attended'] == 1
didnotattend = data['attended'] == 0

attend = data[attendedclass]
no_attend = data[didnotattend]

print('attended class:')
print(attend.mean())
print('')

print('did not attend class:')
print(no_attend.mean())

attended class:
quiz_total     72.500000
attended        1.000000
final_grade    95.072121
final_exam     40.636364
dtype: float64

did not attend class:
quiz_total     68.326087
attended        0.000000
final_grade    91.086159
final_exam     39.442029
dtype: float64


* Those who attended class got higher scores than those who didn't attend class.
* For those who attended class, the average quiz_total, final_grade, and final_exam were 72.5, 95.072121, and 40.636364.
* For those who didn't attend class, the average quiz_total, final_grade, and final_exam were 68.326087, 91.086159, and 39.442029.

# **Block 4: For 1 bonus point of extra credit: based on the data “quiz_total”, “final_grade”, “final_exam” you have, was anyone working together? If so, spit out their names.**

In [ ]:
work_together = data[data.duplicated(subset = ['quiz_total', 'final_grade', 'final_exam'], keep = False)]
print(work_together)

     quiz_total  attended  final_grade  final_exam           name
146        72.0         0        92.81          38  Jaraham Eidda
170        72.0         0        92.81          38  Addie Maharaj


* Jaraham Eidda and Addie Maharaj worked together